### PR TITLE
Remove legacy schema introspection

### PR DESCRIPTION
**Pull request by SimonSapin**
Schema introspection now runs natively without involving JavaScript code. We have high confidence that the two implementations return matching responses based on differential testing fuzzing arbitrary queries against a large schema, and testing a corpus of customer schemas against a comprehensive query.

In Router YAML configuration:

* The `experimental_introspection_mode` key is removed, `new` is now the only behavior
* The `supergraph.query_planning.legacy_introspection_caching` key is removed, the behavior is now similar to what was `false`: introspection responses are not part of the query plan cache but instead in a separate, small, in-memory only cache.

Migrations ensure that existing configuration files will keep working.

Fixes ROUTER-338

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Existing relevant tests passing[^3]
    - [x] Unit Tests
    - [x] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Comment by lrlna** [comment]
If introspection is now completely in rust, does that mean v8 won't start up with just `experimental_query_planner_mode: new`? If so, we need to update a few more docs [here](https://github.com/apollographql/router/blob/9e3219c603657eb17ed351c79349c01decd06686/docs/source/configuration/overview.mdx?plain=1#L592-L599) and [here](https://github.com/apollographql/router/blob/9e3219c603657eb17ed351c79349c01decd06686/docs/source/executing-operations/native-query-planner.mdx?plain=1#L35-L40). It seems the docs we serve are now just on `next` branch 🤔 🤔 🤔 so I guess doc changes have to be PR'ed separately to `next`?

File: apollo-router/src/query_planner/bridge_query_planner.rs [hunk 1]
```diff
@@ -236,27 +224,6 @@ impl PlannerMode {
...
-    ) -> Result<Arc<Introspection>, ServiceBuildError> {
-        let js_planner = match self {
-            Self::Js(js) => js.clone(),
-            Self::Both { js, .. } => js.clone(),
-            Self::Rust(_) => {
-                // JS "planner" (actually runtime) was not created for planning
-                // but is still needed for introspection, so create it now
-                Self::js_planner(sdl, configuration, old_js_planner).await?
-            }
-        };
```
Comment (SimonSapin) [inline apollo-router/src/query_planner/bridge_query_planner.rs:254]: > If introspection is now completely in rust, does that mean v8 won't start up with just `experimental_query_planner_mode: new`? 

Correct. This is the relevant code change. Here the badly-named `js_planner` represents a V8 runtime.

I didn’t realize we publish docs from `next`, I’ll make a separate PR

File: apollo-router/tests/integration/redis.rs [hunk 2]
```diff
@@ -959,26 +978,6 @@ async fn query_planner_redis_update_planner_mode() {
     .await;
 }
 
-#[tokio::test(flavor = "multi_thread")]
-async fn query_planner_redis_update_introspection() {
```
Comment (SimonSapin) [inline apollo-router/tests/integration/redis.rs:963]: This test is removed because it specifically tests that enabling introspection in config changes Redis cache keys. This PR changes `caching_query_planner.rs` for this to no longer be the case, since introspection responses (or "introspection is disabled" errors) are now never in this cache.

File: apollo-router/src/introspection.rs [hunk 3]
```diff
@@ -1,142 +1,163 @@
...
-                    incremental_delivery: Some(IncrementalDeliverySupport {
-                        enable_defer: Some(true),
-                    }),
-                    graphql_validation: true,
-                    reuse_query_fragments: Some(false),
-                    generate_query_fragments: None,
-                    debug: None,
-                    type_conditioned_fetching: false,
-                },
+    fn execute_intropsection(schema: &spec::Schema, doc: &ParsedDocument) -> graphql::Response {
```
Comment (Geal) [inline apollo-router/src/introspection.rs:141]:
```suggestion
    fn execute_introspection(schema: &spec::Schema, doc: &ParsedDocument) -> graphql::Response {
```

**Review by Geal — APPROVED** [review]
thank you, it's great to see that code finally removed!

File: docs/source/configuration/overview.mdx [hunk 4]
```diff
@@ -610,20 +610,6 @@ The default value of `experimental_parallelism` is `1`.
 
 In practice, you should tune `experimental_parallelism` based on metrics and benchmarks gathered from your router.
 
-<MinVersion version="1.51.0">
-
-### Introspection response caching
-
-</MinVersion>
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply docs/source/configuration/overview.mdx:617]: Not as far as I can see

File: apollo-router/src/introspection.rs (hunk 3 above)
Comment (SimonSapin) [reply apollo-router/src/introspection.rs:141]: I seem to like that particular typo, I made it in two places
